src/code/z_scene_table.c

```diff
@@ -30,784 +30,1738 @@
+    ENTRANCE(SCENE_LINK_HOME, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_LINK_HOME, 0, false, true, TRANS_TYPE_CS_BLACK_FILL, TRANS_TYPE_CS_BLACK_FILL),
+    ENTRANCE(SCENE_KOKIRI_SHOP, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK_FAST),
+    ENTRANCE(SCENE_KOKIRI_SHOP, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK_FAST),
+    ENTRANCE(SCENE_KOKIRI_SHOP, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK_FAST),
+    ENTRANCE(SCENE_KOKIRI_SHOP, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK_FAST),
+    ENTRANCE(SCENE_DDAN, 1, true, false, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_DDAN, 1, true, false, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_DDAN, 1, true, false, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_DDAN, 1, true, false, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_KOKIRI_HOME, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK_FAST),
+    ENTRANCE(SCENE_KOKIRI_HOME, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK_FAST),
+    ENTRANCE(SCENE_KOKIRI_HOME, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK_FAST),
+    ENTRANCE(SCENE_KOKIRI_HOME, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK_FAST),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_FADE_WHITE, TRANS_TYPE_FADE_WHITE),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_FADE_WHITE, TRANS_TYPE_FADE_WHITE),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_FADE_BLACK_FAST, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_CIRCLE(TCA_NORMAL, TCC_BLACK, TCS_SLOW),
+             TRANS_TYPE_CIRCLE(TCA_NORMAL, TCC_BLACK, TCS_SLOW)),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_CIRCLE(TCA_NORMAL, TCC_BLACK, TCS_SLOW),
+             TRANS_TYPE_CIRCLE(TCA_NORMAL, TCC_BLACK, TCS_SLOW)),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_CIRCLE(TCA_WAVE, TCC_BLACK, TCS_SLOW),
+             TRANS_TYPE_CIRCLE(TCA_WAVE, TCC_BLACK, TCS_SLOW)),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_FADE_WHITE, TRANS_TYPE_FADE_WHITE),
+    ENTRANCE(SCENE_SPOT00, 0, true, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_CIRCLE(TCA_WAVE, TCC_BLACK, TCS_SLOW),
+             TRANS_TYPE_CIRCLE(TCA_WAVE, TCC_BLACK, TCS_SLOW)),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_FADE_WHITE, TRANS_TYPE_FADE_WHITE),
+    ENTRANCE(SCENE_SPOT00, 0, false, true, TRANS_TYPE_CIRCLE(TCA_NORMAL, TCC_WHITE, TCS_SLOW),
+             TRANS_TYPE_CIRCLE(TCA_NORMAL, TCC_WHITE, TCS_SLOW)),
```

Comment:
Mostly just a note but we will probably have to think of a shorter way to define these entries (maybe shortening the enums or working on the format), since these can get pretty long. Plus they would be even longer with the entrance enum and offsets if the table was made into its own include file

Reply:
agreed. I tried my best to make them short while also keeping prefixes on the enums. If there are any ideas on how to change them would be interested to hear

Reply:
Overall, I don't think there's a way around the verbosity, there are just many parameters.

Pushing one of my previous suggestions to the extreme:

<details>

```c
#define TCA_R TCA_RIPPLE
#define TCC_W TCC_WHITE
#define TCS_F TCS_FAST

#define TRANS_TYPE_C(a, c, s) TRANS_TYPE_CIRCLE(TCA_##a, TCC_##c, TCS_##s)

#define ENTRANCE(scene, spawn, continueBgm, displayTitleCard, fadeIn, fadeOut)                                   \
    {                                                                                                            \
        (scene), (spawn),                                                                                        \
            (((continueBgm)&1) << 15) | (((displayTitleCard)&1) << 14) | (((TRANS_TYPE_##fadeIn) & 0x7F) << 7) | \
                ((TRANS_TYPE_##fadeOut) & 0x7F)                                                                  \
    }

EntranceInfo gEntranceTable[] = {
    ENTRANCE(SCENE_YDAN, 0, false, true, FADE_BLACK, FADE_BLACK),
    ENTRANCE(SCENE_SPOT06, 7, false, true, C(R, W, F), C(R, W, F)),
};
```

</details>

Less extreme:

<details>

```c
#define TRANS_TYPE_TCIRCLE(a, c, s) TRANS_TYPE_CIRCLE(TCA_##a, TCC_##c, TCS_##s)

#define ENTRANCE(scene, spawn, continueBgm, displayTitleCard, fadeIn, fadeOut)                                   \
    {                                                                                                            \
        (scene), (spawn),                                                                                        \
            (((continueBgm)&1) << 15) | (((displayTitleCard)&1) << 14) | (((TRANS_TYPE_##fadeIn) & 0x7F) << 7) | \
                ((TRANS_TYPE_##fadeOut) & 0x7F)                                                                  \
    }

EntranceInfo gEntranceTable[] = {
    ENTRANCE(SCENE_YDAN, 0, false, true, FADE_BLACK, FADE_BLACK),
    ENTRANCE(SCENE_SPOT06, 7, false, true, TCIRCLE(RIPPLE, WHITE, FAST), TCIRCLE(RIPPLE, WHITE, FAST)),
};
```

</details>

Also, easy space gain, replace true/false with 0/1 :)